heron/schedulers/src/java/com/twitter/heron/scheduler/kubernetes/KubernetesController.java

```diff
@@ -15,25 +15,36 @@
 package com.twitter.heron.scheduler.kubernetes;
 
 import java.net.HttpURLConnection;
-import java.util.Arrays;
 import java.util.logging.Level;
 import java.util.logging.Logger;
 
-import com.twitter.heron.spi.utils.NetworkUtils;
+import com.fasterxml.jackson.databind.JsonNode;
+
+import com.twitter.heron.scheduler.TopologySubmissionException;
+import com.twitter.heron.scheduler.utils.SchedulerJsonAPIUtils;
 
 public class KubernetesController {
   private static final Logger LOG = Logger.getLogger(KubernetesController.class.getName());
 
   private final String kubernetesURI;
+  private final String kubernetesNamespace;
   private final String topologyName;
   private final boolean isVerbose;
 
   public KubernetesController(
       String kubernetesURI,
+      String kubernetesNamespace,
       String topologyName,
       boolean isVerbose
   ) {
     this.kubernetesURI = kubernetesURI;
+
+    if (kubernetesNamespace == null) {
+      LOG.log(Level.INFO, "Namespace not provided in configuration. Using default.");
+      this.kubernetesNamespace = "default";
+    } else {
+      this.kubernetesNamespace = kubernetesNamespace;
```

Comment:
Instead of initializing namespace, can we initialize `"%s/api/v1/namespaces/%s/pods` as `baseUriPath`? Every method uses this as their base, so this would reduce duplicate logic.